satellite/satellitedb/overlaycache.go

```diff
@@ -469,7 +447,7 @@ func (cache *overlaycache) UpdateStats(ctx context.Context, updateReq *overlay.U
 }
 
 // UpdateOperator updates the email and wallet for a given node ID for satellite payments.
-func (cache *overlaycache) UpdateOperator(ctx context.Context, nodeID storj.NodeID, operator pb.NodeOperator) (stats *overlay.NodeStats, err error) {
+func (cache *overlaycache) UpdateOperator(ctx context.Context, nodeID storj.NodeID, operator pb.NodeOperator) (stats *overlay.NodeDossier, err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	tx, err := cache.db.Open(ctx)
```

Comment:
I think we can remove transactions here completely.

Reply:
Not really. UpdateOperator may be invoked only with the new Wallet info and `nil` for Email.

We need the transaction, but make better use of it. I will push a fix.

Reply:
Well, actually, you are right. We don't need the transaction, but we also need to fix handing nil values.

Reply:
Wallet and Email are just strings and not pointers. So I checked all places where UpdateOperator is invoked and made sure both Email and Wallet are provided for the update. This way we won't lose one of the values if providing only the other for the update.

I also removed the transaction.